### PR TITLE
[RHELC-1286, RHELC-1555] Add tests validating the ELS support

### PR DESCRIPTION
**Pull request by danmyway**
* add a full conversion test using the ELS repositories
* add non-destructive scenarios validating a correct repository enablement
* slightly modify the eus_support test

Jira Issues:

- [RHELC-1286](https://issues.redhat.com/browse/RHELC-1286)
- [RHELC-1555](https://issues.redhat.com/browse/RHELC-1555)

Checklist

- [x] PR has been tested manually in a VM (either author or reviewer)
- [x] Jira issue has been made public if possible
- [x] `[RHELC-]` is part of the PR title 
- [x] GitHub label has been added to help with Release notes 
- [x] PR title explains the change from the user's point of view
- [x] Code and tests are documented properly
- [x] The commits are squashed to as few commits as possible (without losing data) 
- [x] When merged: Jira issue has been updated to `Release Pending` if relevant

### CODE REVIEW

File: tests/integration/tier0/non-destructive/els/main.fmf [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            The --els option is used in the command. Only regular (non-els) repoids should be enabled.
+        2/ The start date of the ELS phase is set far to the past to simulate the ELS phase began.
+            The --els option *is used*.
+            The report is expected to print out a WARNING and advise to use the --els option.
+    tag+:
+        - test-els-support
+    test: |
+      pytest -m test_els_support
+
+/non_els_account:
```
Comment (kokesak) [inline tests/integration/tier0/non-destructive/els/main.fmf:30]: Add `result: xfail` to this

File: tests/integration/tier0/non-destructive/els/test_els_support.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+            assert shell(f"mv -v {backup} {file}").returncode == 0
+
+    _restore_els_mapping()
+
+
+REGULAR_REPOID_MSG = "RHEL repository IDs to enable: rhel-7-server-rpms"
+ELS_REPOID_MSG = "RHEL repository IDs to enable: rhel-7-server-els-rpms"
+
+MOCKED_ELS_PHASE = '"1970-01-01"'
+MOCKED_NON_ELS_SYSVER = "if system_info.version.major == 42:"
```
Comment (kokesak) [inline tests/integration/tier0/non-destructive/els/test_els_support.py:45]: no 420? :cry:

File: tests/integration/tier0/non-destructive/els/test_els_support.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+    def _restore_els_mapping():
+        for file, backup in file_backup_mapping.items():
+            assert shell(f"mv -v {backup} {file}").returncode == 0
+
+    _restore_els_mapping()
+
+
+REGULAR_REPOID_MSG = "RHEL repository IDs to enable: rhel-7-server-rpms"
+ELS_REPOID_MSG = "RHEL repository IDs to enable: rhel-7-server-els-rpms"
```
Comment (kokesak) [inline tests/integration/tier0/non-destructive/els/test_els_support.py:42]: this seems unused

File: tests/integration/tier0/non-destructive/els/test_els_support.py (hunk 2 above)
Comment (danmyway) [reply tests/integration/tier0/non-destructive/els/test_els_support.py:45]: My 42 got overwritten, so I need to compensate here

File: tests/integration/tier0/non-destructive/els/test_els_support.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+                )
+            c2r.expect("Repositories enabled through subscription-manager", timeout=120)
+            c2r.sendcontrol("c")
+
+
+@pytest.mark.test_rhsm_non_els_account
+def test_rhsm_non_els_account(convert2rhel):
+    """
+    Verify that Convert2RHEL is working properly when ELS repositories are not available for conversions
+    (the account does not have the ELS SKU available) to RHEL ELS minor versions (8.6, ...)
```
Comment (kokesak) [inline tests/integration/tier0/non-destructive/els/test_els_support.py:111]:
```suggestion
    (the account does not have the ELS SKU available) to RHEL 7 ELS 
```

File: tests/integration/tier0/non-destructive/els/test_els_support.py [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+els_support_parameters = [
+    # System not recognized as an ELS candidate with --els option used
+    ("--els", REGULAR_REPOID_MSG, None, MOCKED_NON_ELS_SYSVER, False, False),
+    # System recognized as an ELS candidate after the start of ELS phase without --els option
+    ("", REGULAR_REPOID_MSG, MOCKED_ELS_PHASE, None, False, True),
+]
+
+
+@pytest.mark.parametrize(
+    "additional_option, repoid_message, modified_startdate, modified_releasever, is_els, recommend_els_msg_displayed",
```
Comment (kokesak) [inline tests/integration/tier0/non-destructive/els/test_els_support.py:57]: Do we need the `is_els` parameter if there is no test case where it is set to `True`?

File: plans/tier0.fmf [hunk 6]
```diff
@@ -73,6 +73,17 @@ description+: |
                 test+<:
                     - conversion-method/rhsm_eus_account_conversion
 
+        /rhsm_ELS_conversion:
```
Comment (r0x0d) [inline plans/tier0.fmf:76]:
```suggestion
        /rhsm_els_conversion:
```

File: tests/integration/common/checks-after-conversion/test_enabled_repositories.py [hunk 7]
```diff
@@ -21,22 +21,38 @@ def _check_enabled_repos_rhel8(enabled_repos: str = "", eus: bool = False):
...
     """
 
     try:
         enabled_repos = shell("yum repolist").output
         system_release = SystemInformationRelease()
         assert "redhat" in system_release.distribution
 
-        if system_release.version.major == 7 and system_release.version.minor == 9:
-            assert "rhel-7-server-rpms/7Server/x86_64" in enabled_repos
+        if system_release.version.major == 7:
```
Comment (r0x0d) [inline tests/integration/common/checks-after-conversion/test_enabled_repositories.py:46]: Pasting the suggestion for this whole if-else section, but what do you think about: 

```suggestion
        if system_release.version.major == 7:
            # Handle the special test case scenario where we use the
            # account with the ELS repositories available
            is_els = os.path.exists("/els_repos_used")
            _check_enabled_repos_rhel7(enabled_repos, els=is_els)
        elif system_release.version.major == 8:
            # Handle the special test case scenario where we use the
            # account with the EUS repositories available
            is_eus = os.path.exists("/eus_repos_used")
            _check_enabled_repos_rhel7(enabled_repos, els=is_eus)
```

Comment (danmyway) [reply tests/integration/common/checks-after-conversion/test_enabled_repositories.py:46]: Smart! I'll do that, thank you, @r0x0d

**Pull request merged**